build.sbt

```diff
@@ -35,6 +35,15 @@ lazy val commonSettings = Seq(
   }
 )
 
+lazy val fatalWarningsSettings = Seq(
+  scalacOptions ++= {
+    CrossVersion.partialVersion(scalaVersion.value) match {
+      case Some((2, n)) if n >= 13 => "-Werror" :: Nil
```

Comment:
I would recommend only enabling this in CI. While you are developing, you might want to leave some warnings unaddressed until you have solved the problem you are working on.
In `chiseltest` we check for warnings like this:
https://github.com/ucb-bar/chiseltest/blob/fed4822cc562293e373ed568d6a12139b631e94c/.github/workflows/test.yml#L159

Reply:
Good idea @ekiwi, I agree that we should do it this way.

Reply:
That seems like a good idea. I will try and get this working in chisel3